### DISCUSSION TITLE
Is there function to create a json from a string looks like this "sys.exec.out.dyn.cannon.pos[1]"

### DISCUSSION THREAD

**Discussion by yusufcananar** [nlohmann/json#4293] [Q&A] [answered]
I am coding a db recording app in C++ and I want to have a json like this 
```
{ "sys" : 
    { "exec" : 
        { "out" : 
           { "time" : "0.65" }, 
           { "dyn" : 
               { "cannon" : 
                   { "pos" : [ "98", "34" ] } 
               } 
           } 
         } 
     }
}
```

Where  I use my own function to record to db:
```
addField(doc, "sys.exec.out.time", "0.65"); 
addField(doc, "sys.exec.out.dyn.cannon.pos[0]", "98"); 
addField(doc, "sys.exec.out.dyn.cannon.pos[1]", "34");
```
currently i am parsing the string by dots and brackets to achieve my goal. 
I want to know if Is there function to create a json from a string looks like this "sys.exec.out.dyn.cannon.pos[1]"?

**Comment by gregmarr** [comment, marked as answer]
You can do this using JSON Pointer and some simple replacements.
https://www.godbolt.org/z/MhafPGhqG

**Reply by yusufcananar** [reply]
You are life saver thank you very much.

**Reply by gregmarr** [reply]
You're welcome.
